src/services/datastore/BlockHourFinder.ts

```diff
@@ -0,0 +1,54 @@
+import { Loggable } from '../../lib/Loggable';
+import { IBlockHoursFinder } from '../../interfaces/datastore/IBlockHoursFinder';
+import { FindBlockHourResult } from '../../types';
+import { ILogger } from '../../interfaces/lib/ILogger';
+import { IAlerts } from '../../interfaces/lib/IAlerts';
+import { BlockHour } from '../../entities/BlockHour';
+import { IDataStore } from '../../interfaces';
+import { IRepository } from '../../interfaces/datastore/IRepository';
+
+export class BlockHourFinder extends Loggable implements IBlockHoursFinder {
+  protected datastore!: IDataStore;
+
+  /**
+   * @param {IDataStore} datastore typeorm DataSource wrapper
+   * @param {ILogger} logger property used for delegation of routing logs
+   * @param {IAlerts} alerts property used for delegation of sending alerts
+   */
+  constructor(datastore: IDataStore, logger?: ILogger, alerts?: IAlerts) {
+    super(logger, alerts);
+    this.datastore = datastore;
+  }
+
+  /**
+   * Finds the latest BlockHour in the data store.
+   */
+  async apply(): Promise<FindBlockHourResult> {
+    // resolve repository
+    const repository: IRepository | undefined = this.datastore.getRepository(BlockHour);
+    if (!repository) {
+      const message = 'failed to resolve BlockHour Repository';
+      this.error(message);
+      return { success: false, message };
+    }
+
+    try {
+      // find block entity for given block number
+      const blockHours: BlockHour[] = await repository.find({ order: { id: 'DESC' } });
```

Comment:
May want to limit the query to 1, like this:

```suggestion
      const blockHours: BlockHour[] = await repository.find({
        order: { id: 'DESC' },
        take: 1
      });
```
Otherwise we may be querying the whole table unnecessarily.
I guess `findOne` or `findOneBy` would also do the trick.

Reply:
Also, probably want to sort by a column that guarantees the time series. `hourlyTimestampUtc` is a good candidate, but actually any column other than `id` would suffice.